app.go

```diff
@@ -127,3 +130,28 @@ func api(w http.ResponseWriter, req *http.Request) {
 	}
 	json.NewEncoder(w).Encode(data)
 }
+
+type healthState struct {
+	StatusCode int
+}
+
+var currentHealthState = healthState{200}
+var mutexHealthState = &sync.Mutex{}
+
+func healthHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodPost {
+		var statusCode int
+		err := json.NewDecoder(req.Body).Decode(&statusCode)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+		} else {
+			fmt.Printf("Update health check status code [%d]\n", statusCode)
+			mutexHealthState.Lock()
```

Comment:
It's better to directly call `defer mutexHealthState.Unlock()` right after calling `mutexHealthState.Lock()`.